crates/octez/src/async/client.rs

```diff
@@ -159,6 +159,74 @@ impl TryFrom<StdOut> for Address {
     }
 }
 
+#[derive(Default)]
+struct TransferOptionsBuilder {
+    from: Option<String>,
+    to: Option<String>,
+    amount: Option<u64>,
+    entrypoint: Option<String>,
+    arg: Option<String>,
+    burn_cap: Option<f64>,
+}
+
+impl TransferOptionsBuilder {
+    pub fn new() -> Self {
+        TransferOptionsBuilder {
+            ..Default::default()
+        }
+    }
+
+    pub fn set_from(mut self, from: String) -> Self {
+        self.from = Some(from);
+        self
+    }
+
+    pub fn set_to(mut self, to: String) -> Self {
+        self.to = Some(to);
+        self
+    }
+
+    pub fn set_entrypoint(mut self, entrypoint: String) -> Self {
+        self.entrypoint = Some(entrypoint);
+        self
+    }
+
+    pub fn set_arg(mut self, arg: String) -> Self {
+        self.arg = Some(arg);
+        self
+    }
+
+    pub fn set_amount(mut self, amount: u64) -> Self {
+        self.amount = Some(amount);
+        self
+    }
+
+    pub fn set_burn_cap(mut self, burn_cap: f64) -> Self {
+        self.burn_cap = Some(burn_cap);
+        self
+    }
+
+    pub fn build(self) -> Result<TransferOptions> {
+        Ok(TransferOptions {
+            from: self.from.ok_or(anyhow!("Missing from"))?,
+            to: self.to.ok_or(anyhow!("Missing to"))?,
+            amount: self.amount.unwrap_or_default(),
+            entrypoint: self.entrypoint,
+            arg: self.arg,
+            burn_cap: self.burn_cap,
+        })
+    }
+}
+
+struct TransferOptions {
+    from: String,
+    to: String,
+    amount: u64,
```

Comment:
This needs to be `f64` since `amount` is in tez and fractions are accepted.

Reply:
good point, thanks!